.circleci/config.yml

```diff
@@ -114,6 +117,65 @@ jobs:
               docker push ${CIRCLE_PROJECT_USERNAME}/${CIRCLE_PROJECT_REPONAME}:${CIRCLE_TAG}
             fi
 
+  proxy-build:
+    executor: docker/machine
+    steps:
+      - checkout
+      - run:
+          name: Generate build version.json
+          command: >
+            printf
+            '{"commit":"%s","version":"%s","source":"https://github.com/%s/%s","build":"%s"}\n'
+            "$CIRCLE_SHA1" "$CIRCLE_TAG" "$CIRCLE_PROJECT_USERNAME"
+            "$CIRCLE_PROJECT_REPONAME" "$CIRCLE_BUILD_URL"
+            > version.json
+      - run:
+          name: Output version.json
+          command: cat version.json
+      - docker/build:
+          path: dockerfiles/proxy
+          image: ${CIRCLE_PROJECT_REPONAME}-auth0-proxy
+          tag: ${CIRCLE_SHA1:0:9}
+      - run:
+          name: docker save
+          command: |
+            mkdir -p /tmp/cache
+            docker save -o /tmp/cache/auth0-proxy.tar "auth0-proxy:build"
+      - persist_to_workspace:
+          root: /tmp/cache
+          paths:
+            - auth0-proxy.tar
```

Comment:
This is used when we want to build once and use the docker image multiple times, i.e. for downstream tests and push jobs. In this case, we can combine build and publish in a single step. This code can be removed.